Sources/Signal.swift

```diff
@@ -67,7 +67,7 @@ public final class Signal<Value, Error: Swift.Error> {
 			state = .alive(Bag(), hasDeinitialized: false)
 
 			stateLock = Lock.make()
-			sendLock = Lock.make()
+			sendLock = Lock.PthreadLock(recursive: true)
```

Comment:
Agree with this change. Ref: https://github.com/ReactiveCocoa/ReactiveSwift/issues/74#issuecomment-472686906